jwql/utils/crds_tools.py

```diff
@@ -0,0 +1,190 @@
+    crds_dict['READPATT'] = yaml_dict['Readout']['readpatt'].upper()
+
+    # Currently, all reference files that use SUBARRAY as a selection
+    # criteria contain SUBARRAY = 'GENERIC', meaning that SUBARRAY
+    # actually isn't important. So let's just set it to FULL here.
+    crds_dict['SUBARRAY'] = 'FULL'
+
+    # Use the current date and time in order to get the most recent
+    # reference file
+    crds_dict['DATE-OBS'] = datetime.date.today().isoformat()
+    current_date = datetime.datetime.now()
+    crds_dict['TIME-OBS'] = current_date.time().isoformat()
+
+    array_name = yaml_dict['Readout']['array_name']
+    crds_dict['DETECTOR'] = array_name.split('_')[0].upper()
+    if '5' in crds_dict['DETECTOR']:
+        crds_dict['DETECTOR'] = crds_dict['DETECTOR'].replace('5', 'LONG')
+
+    if 'FGS' in crds_dict['DETECTOR']:
+        crds_dict['DETECTOR'] = 'GUIDER{}'.format(crds_dict['DETECTOR'][-1])
+
+    if instrument == 'NIRCAM':
+        if crds_dict['DETECTOR'] in ['NRCALONG', 'NRCBLONG']:
+            crds_dict['CHANNEL'] = 'LONG'
+        else:
+            crds_dict['CHANNEL'] = 'SHORT'
+
+    # For the purposes of choosing reference files, the exposure type should
+    # always be set to imaging, since it is used to locate sources in the
+    # seed image, prior to any dispersion.
+    crds_dict['EXP_TYPE'] = EXPTYPES[instrument.lower()]["imaging"]
+
+    # This assumes that filter and pupil names match up with reality,
+    # as opposed to the more user-friendly scheme of allowing any
+    # filter to be in the filter field.
+    crds_dict['FILTER'] = yaml_dict['Readout']['filter']
+    crds_dict['PUPIL'] = yaml_dict['Readout']['pupil']
+    return crds_dict
+
+
+def get_reffiles(parameter_dict, reffile_types, download=True):
+    """Determine CRDS's best reference files to use for a particular
+    observation, and download them if they are not already present in
+    the CRDS_PATH. The determination is made based on the information in
+    the parameter_dictionary.
+
+    Parameters
+    ----------
+
+    parameter_dict : dict
+        Dictionary of basic metadata from the file to be processed by the
+        returned reference files (e.g. INSTRUME, DETECTOR, etc)
+
+    reffile_types : list
+        List of reference file types to look up and download. These must
+        be contained in CRDS's list of reference file types.
+
+    download : bool
+        If True (default), the identified best reference files will be
+        downloaded. If False, the dictionary of best reference files will
+        still be returned, but the files will not be downloaded. The use
+        of False is primarily intended to support testing on Travis.
+
+    Returns
+    -------
+
+    reffile_mapping : dict
+        Mapping of downloaded CRDS file locations
+    """
+    # IMPORTANT: Import of crds package must be done AFTER the environment
+    # variables are set in the functions above
+    import crds
```

Comment:
I think this is a new dependency for the project.  Could you add the appropriate lines to `setup.py`, `requirements.txt`, and `environment-3.6.yml`?

Reply:
Done